### ISSUE TITLE
Operate on shipped FSharp.Core binaries

### ISSUE THREAD

**Issue by cartermp** [fsharp/docs#31]
Once a new FSharp.Core is released with some of the core docs improvements this repo should only use the latest built FSharp.Core. There should be an advanced option for seeing what it _would_ look like with a new FSharp.Core where you can clone down latest master of dotnet/fsharp and build the docs from there.

Moreover, this is the only sane way we can generated versioned documentation so that folks who must use an older F# and FSharp.Core can see the documentation that they need.

**Comment by cartermp** [comment]
We may also want to consider operating on the latest myget binary: https://dotnet.myget.org/feed/fsharp/package/nuget/FSharp.Core

This should in theory have all the latest improvements to docs before they actually ship, and if not we should work to make that happen rather than require building from source

**Comment by dsyme** [comment]
Agreed

**Comment by dsyme** [comment]
BTW one issue with this is that the source code paths in the source range locations in the FSharp.Core shipped binaries include paths like `F:\1\work\something`.  Without knowing (or removing at build-time) this path we can't give accurate github source code links 

By building `fsharp` locally we can set `--sourcepath fsharp` and the FSHarp.Formtting ApiDocs generation does its thing.

**Comment by baronfel** [comment]
One nice thing about supporting the deterministic builds ( and there's an open issue on the f sharp compiler about this) is that that root path becomes a constant that can be stripped out. that makes jobs like this, and source link overall, a lot more consistent to deal with.

**Comment by baronfel** [comment]
Here's the issue I'm referring to: https://github.com/dotnet/fsharp/issues/9379

**Comment by dsyme** [comment]
@baronfel It's kind of good to sit on this side of the fence for a while, to see how these issues bite in practice 😄 

> Here's the issue I'm referring to: dotnet/fsharp#9379

Cool thanks, there's also this one: https://github.com/dotnet/fsharp/issues/8883, looks like the same issue?

**Comment by baronfel** [comment]
at a first glance it certainly looks similar. If it's related to that path mapping then it sounds like it's going to be a problem in the core targets.

**Comment by cartermp** [comment]
Yeah, we need a core targets change or we need to move these targets (via copy pasta) directly into the F# ones. I think we should entertain the latter given that core pieces of .NET are starting to lock down for .NET 5 now

**Comment by baronfel** [comment]
For that effort, would you want a copy paste to a c# helper assembly, or reimplementation on F# in the existing FSharp.Build.dll?

**Comment by cartermp** [comment]
My understanding is that it's just an MSBuild target, so it could presumably be added to the equivalent place it's used in C# targets. I think it would be best to consult @kevinransom on that though, since he's the author of the F# targets

**Comment by dsyme** [comment]
Closing this as duplicate of #75
